### DISCUSSION TITLE
Get indentation level of bullets?

### DISCUSSION THREAD

**Discussion by mad-moo** [ShapeCrawler/ShapeCrawler#376] [Ideas]
Does ShapeCrawler support getting the level of a bullet point? I can see size, character and type of the bullet, but not how far it is actually indented.

![image](https://user-images.githubusercontent.com/24632992/200165966-7bd6d04c-ddc2-44f3-9d7c-dc0870a66247.png)

**Comment by AdamShakhabov** [comment]
No, it is not supported yet.

I've created https://github.com/ShapeCrawler/ShapeCrawler/issues/377 issue to implement this.

**Reply by mad-moo** [reply]
Thank you!

**Comment by DarwinRajagopalan** [comment]
@adamshakhabov  , @huchim   

**How can i create tree structure bullet list**

For Example : 
![image](https://github.com/ShapeCrawler/ShapeCrawler/assets/85544473/622637d5-eacd-41f0-8900-2511e76bdfae)



**### Below link there is no example for the  sub list bullet creation** 
https://github.com/ShapeCrawler/ShapeCrawler/wiki/Examples#addupdate-paragraph-bullet 

 

**_Where am in now_** 
```
var bullet = textFrame!.Paragraphs.Add().Bullet;
bullet.Type = SCBulletType.Character;
bullet.Character = "•";
bullet.Size = 100;
textFrame!.Paragraphs.Last().AddPortion("Coffee");
textFrame!.Paragraphs.Last().AddPortion("Tea"); // below I have to create sub bullet list with different character symbol 
                    textFrame!.Paragraphs.Last().AddPortion("Black Tea"); // sub 1
                    textFrame!.Paragraphs.Last().AddPortion("Green Tea");// sub 2
textFrame!.Paragraphs.Last().AddPortion("Milk");
```

**Reply by DarwinRajagopalan** [reply]
@huchim kindly help me on this .. i have to show the demo to client on coming Monday.  almost everything i have done by your plugin ..

**Reply by AdamShakhabov** [reply]
@DarwinRajagopalan, check this beta version [**v0.47.0-beta.1**](https://github.com/ShapeCrawler/ShapeCrawler/files/11985365/ShapeCrawler.0.47.0-beta.1.nupkg.zip). Setter for `IParagraph.IndentLevel` was added:

```c#
var pres = SCPresentation.Open(@"c:\temp\test.pptx");
var paragraphs = pres.Slides[0].Shapes.GetByName<IAutoShape>("TextBox 1").TextFrame!.Paragraphs;
var para1 = paragraphs.Add();
para1.Text = "Tea";
var para2 = paragraphs.Add();
para2.Text = "Black tea";

para2.IndentLevel = 2;

pres.Save();
```
